src/test/java/spoon/test/pkg/testclasses/ElementProcessor.java

```diff
@@ -0,0 +1,17 @@
+package spoon.test.pkg.testclasses;
+
+import spoon.processing.AbstractProcessor;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtPackage;
+
+/**
+ * Created by urli on 11/09/2017.
+ */
+public class ElementProcessor extends AbstractProcessor<CtElement> {
+    public void process(CtElement element) {
+        if(element instanceof CtPackage && element != getFactory().Package().getRootPackage()){
```

Comment:
there is a confusion, the goal is that client code indeed never has to write this. 

so the implementation RootPackage.setSimpleName would simply be ... empty.

do you see what I mean?

Reply:
> do you see what I mean?

Yes, I just kept going on my initial idea. I'll change that.